backend/app/models.py

```diff
@@ -40,15 +42,62 @@ class TimeStampMixin(SQLModel):
     )
 
 
+class DistrictrMap(TimeStampMixin, SQLModel, table=True):
+    uuid: str = Field(sa_column=Column(UUIDType, unique=True, primary_key=True))
+    name: str | None = Field(nullable=False)
+    # This is intentionally not a foreign key on `GerryDBTable` because in some cases
+    # this may be the GerryDBTable but in others the pop table may be a materialized
+    # view of two GerryDBTables in the case of shatterable maps.
+    # We'll want to enforce the constraint tha the gerrydb_table_name is either in
+    # GerrydbTable.name or a materialized view of two GerryDBTables some other way.
+    gerrydb_table_name: str | None = Field(nullable=True)
+    # Null means default number of districts? Should we have a sensible default?
+    num_districts: int | None = Field(nullable=True, default=None)
+    tiles_s3_path: str | None = Field(nullable=True)
+    parent_layer: str = Field(
+        sa_column=Column(UUIDType, ForeignKey("gerrydbtable.uuid"), nullable=False)
+    )
+    child_layer: str | None = Field(
+        sa_column=Column(
+            UUIDType, ForeignKey("gerrydbtable.uuid"), default=None, nullable=True
+        )
+    )
+    # schema? will need to contrain the schema
+    # where does this go?
+    # when you create the view, pull the columns that you need
+    # we'll want discrete management steps
+
+
+class DistrictrMapCreate(BaseModel):
+    name: str
+    gerrydb_table_name: str | None = None
+    num_districts: int | None = None
+    tiles_s3_path: str | None = None
+    parent_layer: str
+    child_layer: str | None = None
+
+
 class GerryDBTable(TimeStampMixin, SQLModel, table=True):
     uuid: str = Field(sa_column=Column(UUIDType, unique=True, primary_key=True))
+    # Must correspond to the layer name in the tileset if a
     name: str = Field(nullable=False, unique=True)
-    tiles_s3_path: str | None = Field(nullable=True)
+
+
+class ParentChildEdges(TimeStampMixin, SQLModel, table=True):
+    id: int = Field(sa_column=Column(Integer, primary_key=True, autoincrement=True))
```

Comment:
(pp) IMO we don't need a primary key here- I'd drop the field.

Reply:
done